hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/TokenKeyCall.java

```diff
@@ -69,10 +82,51 @@ public TokenKeyCall(
     private @NonNull FullResult fullResultsFor(
             @NonNull final ResponseCodeEnum status, final long gasRequirement, @NonNull final Key key) {
         // @Future remove to revert #9069 after modularization is completed
-        if (isStaticCall && status != SUCCESS) {
+        if ((isStaticCall && status != SUCCESS) || status == INVALID_TOKEN_ID || status == KEY_NOT_PROVIDED) {
             return revertResult(status, 0);
         }
         return successResult(
                 TOKEN_KEY.getOutputs().encodeElements(status.protoOrdinal(), keyTupleFor(key)), gasRequirement);
     }
+
+    @Override
+    public @NonNull PricedResult execute() {
```

Comment:
In some suite test cases we compare the returned status represented in string with the expected string representation like [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/CreatePrecompileSuite.java#L260) (more occurrences in the same file) and also in those two cases we would have to fail, so `contractFunctionResultFailedFor()` should be called instead of `contractFunctionResultSuccessFor()`